src/RFC-0303_DanOverview.md

```diff
@@ -0,0 +1,189 @@
+
+### Templates
+
+Templates are parameterised smart contracts. Templates are intended to be well-tested, secure, reusable components 
+for building and running smart contracts on the DAN.
+
+For example, an NFT template would allow a user to populate a few fields, such as name, number of tokens, media 
+locations, and then launch a new NFT series without having to write any actual code.
+
+Templates are stored and managed on the base layer. You can think of the Tari base layer as a type of _git_ for 
+smart contracts. Templates will also have version control features and a smooth upgrade path for existing contracts.
+
+### Contracts
+
+Tari smart contracts are the meat of the DAN ecosystem. Usually a smart contract will be comprised of one or more 
+Tari templates, glue code, and initialisation code.
+
+The contracts are always executed in the Tari Virtual machines. The input and output of every contract instruction 
+is validated by validator nodes that reach consensus using the Cerberus consensus engine.
+
+### Validator Nodes
+
+Validator Nodes (VNs) execute and reach consensus on DAN contract instructions. VNs must register on the base layer 
+and lock up funds (the registration deposit) in order to participate in the DAN. 
+
+With this in place, every base node has an up-to-the-minute list of all active validator nodes and their metadata. The 
+registration deposit also serves as a Sybil prevention mechanism.
+
+Validator nodes are the bridge between the consensus layer and the Tari Virtual Machine (TVM).
+
+VNs are required to re-register periodically as a proof-of-liveness mechanism.
+
+Validator nodes must be able to
+1. interpret the instructions they receive from clients, identifying the contract code that the instruction refers,
+2. retrieve and deserialize the relevant input state,
+3. compute the output state that result from applying the contract logic to the input state, and
+4. reach consensus with its peers.
+
+Steps 1 - 3 are carried out in the Tari Virtual Machine (TVM).
+Step 4 is achieved by communicating with peers via the DAN consensus layer.
+
+[Thaums](#thaums-and-the-turbine-model) exist at the Validator node level, and VNs earn fees, in Thaums, for each 
+instruction that it aids in getting finalised.
+
+### DAN consensus layer
+
+The Cerberus BFT consensus algorithm runs on the consensus layer. This layer is completely ignorant of the 
+semantics of DAN smart contracts.
+
+This layer only cares that:
+* _only_ VNs that have registered on the base layer are participating in consensus.
+* VNs are self-organising into VN committees and are carrying out the rules of Cerberus and emergent HotStuff 
+  correctly.
+
+In particular, the consensus layer has _no idea_ whether an instruction's output is correct. If two-thirds (plus one)
+of the committee say that the results are correct, then consensus has been reached and the consensus layer is happy.
```

Comment:
```suggestion
of the committee agree on the results, then consensus has been reached and the consensus layer is happy.
```

Reply:
It's perhaps also worth mentioning that the result they agree on could be an error in executing the code.